var/spack/repos/builtin/packages/boost/boostorg/variants.py

```diff
@@ -0,0 +1,1479 @@
+            {"when": "~describe", "msg": "Boost.mysql requires Boost.describe"},
+            {"when": "~pfr", "msg": "Boost.mysql requires Boost.pfr"},
+            {"when": "~variant2", "msg": "Boost.mysql requires Boost.variant2"},
+            {"when": "~asio", "msg": "Boost.mysql requires Boost.Asio"},
+            {"when": "@1.85.0: ~charconv", "msg": "Boost.mysql requires Boost.Charconv"},
+        ],
+        description="MySQL client library built on top of Boost.Asio.",
+    )
+    variants.add(
+        "compat",
+        when="@1.83.0:",
+        buildable="@1.87.0:",
+        conflicts=[
+            {"when": "cxxstd=98", "msg": "Boost.compat requires cxxstd >= 11"},
+            {"when": "cxxstd=03", "msg": "Boost.compat requires cxxstd >= 11"},
+        ],
+        description="C++11 implementations of standard components added in later C++ standards",
+    )
+    variants.add(
+        "redis",
+        when="@1.84.0:",
+        buildable="@1.87.0:",
+        default=False,
+        conflicts=[
+            {"when": "cxxstd=11", "msg": "Boost.Redis requires cxxstd >= 17"},
+            {"when": "cxxstd=14", "msg": "Boost.Redis requires cxxstd >= 17"},
+            {"when": "~asio", "msg": "Boost.Redis requires Boost.Asio"},
+        ],
+        description="Redis async client library built on top of Boost.Asio",
+    )
+    variants.add(
+        "cobalt",
+        default=False,
+        when="@1.84.0:",
+        buildable="@1.84.0:",
+        conflicts=[
+            {"when": "cxxstd=11", "msg": "Boost.cobalt requires cxxstd >= 20"},
+            {"when": "cxxstd=14", "msg": "Boost.cobalt requires cxxstd >= 20"},
+            {"when": "cxxstd=17", "msg": "Boost.cobalt requires cxxstd >= 20"},
+            {"when": "~leaf", "msg": "Boost.cobalt requires Boost.leaf"},
+            {"when": "~variant2", "msg": "Boost.cobalt requires Boost.variant2"},
+        ],
+        description=(
+            "Simple single-threaded asynchronicity akin to node.js and asyncio in python"
+        ),
+    )
+    variants.add(
+        "charconv",
+        when="@1.85.0:",
+        buildable="@1.85.0:",
+        conflicts=[
+            {"when": "cxxstd=98", "msg": "Boost.Charconv requires cxxstd >= 11"},
+            {"when": "cxxstd=03", "msg": "Boost.Charconv requires cxxstd >= 11"},
+        ],
+        description="An implementation of C++20's <charconv> in C++11",
+    )
+    variants.add(
+        "scope",
+        when="@1.85.0:",
+        buildable="@1.87.0:",
+        description="Scope guards and a unique_resource wrapper",
+    )
+    variants.add(
+        "parser",
+        when="@1.87.0:",
+        default=False,
+        conflicts=[
+            {"when": "cxxstd=11", "msg": "Boost.Parser requires cxxstd >= 17"},
+            {"when": "cxxstd=14", "msg": "Boost.Parser requires cxxstd >= 17"},
+        ],
+        description="A parser combinator library for building complex parsers",
+    )
+
```

Comment:
```suggestion
    variants.add(
        "mqtt5",
        when="@1.88.0:",
        buildable="@1.88.0:",
        description="A C++17 MQTT client",
    )
```
and possibly `hash2`? It looks like you're modeling all the libraries now, right? Not only the ones that require building with `--with-libraries`?

From Boost 1.88.0: https://github.com/spack/spack/pull/50158.

Reply:
> and possibly `hash2`?

Yup. I just haven't added 1.88 yet.

>It looks like you're modeling all the libraries now, right? Not only the ones that require building with `--with-libraries`?

I model any library that has a constraint. Which subset that is changes by version. For example, in 1.88.0 the Iterator library dropped support for C++03.